cli_test.go

```diff
@@ -0,0 +1,350 @@
+				return
+			}
+		})
+	}
+
+	for name, sig := range niceNameToSigs {
+		hupArg := fmt.Sprintf("--signal-on-hup=%s", name)
+		t.Run(hupArg, func(t *testing.T) {
+			opts, err := c.ParseArgs("ls", hupArg)
+			if !assert.NoError(t, err, "cli.ParseArgs should succeed") {
+				return
+			}
+
+			expected := options{
+				Command:     "ls",
+				Interval:    1,
+				SignalOnHUP: name,
+			}
+
+			if !assert.Equal(t, &expected, opts) {
+				return
+			}
+
+			if err := findInOptionList(t, opts, "signal_on_hup", sig); err != nil {
+				return
+			}
+		})
+		termArg := fmt.Sprintf("--signal-on-term=%s", name)
+		t.Run(termArg, func(t *testing.T) {
+			opts, err := c.ParseArgs("ls", termArg)
+			if !assert.NoError(t, err, "cli.ParseArgs should succeed") {
+				return
+			}
+
+			expected := options{
+				Command:      "ls",
+				Interval:     1,
+				SignalOnTERM: name,
+			}
+
+			if !assert.Equal(t, &expected, opts) {
+				return
+			}
+
+			if err := findInOptionList(t, opts, "signal_on_term", sig); err != nil {
+				return
+			}
+		})
+	}
+
+	for _, i := range []int{5, 10} {
+		arg := fmt.Sprintf("--kill-old-delay=%d", i)
+		t.Run(arg, func(t *testing.T) {
+			opts, err := c.ParseArgs("ls", arg)
+			if !assert.NoError(t, err, "cli.ParseArgs should succeed") {
+				return
+			}
+
+			expected := options{
+				Command:      "ls",
+				Interval:     1,
+				KillOldDelay: intOpt{Valid: true, Value: i},
+			}
+
+			if !assert.Equal(t, &expected, opts) {
+				return
+			}
+			if err := findInOptionList(t, opts, "kill_old_delay", time.Duration(i)*time.Second); err != nil {
+				return
+			}
+		})
+	}
+
+	paths := []string{
+		"/tmp/foo.sock",
+		"/tmp/bar.sock",
+	}
+	for i := 1; i <= 2; i++ {
+		args := make([]string, len(paths))
+		for i, p := range paths {
+			args[i] = "--path=" + p
+		}
+		name := strings.Join(args, " ")
+		t.Run(name, func(t *testing.T) {
+			opts, err := c.ParseArgs(append([]string{"ls"}, args[:i]...)...)
+			if !assert.NoError(t, err, "cli.ParseArgs should succeed") {
+				return
+			}
+
+			expected := options{
+				Command:  "ls",
+				Interval: 1,
+				Paths:    paths[:i],
+			}
+
+			if !assert.Equal(t, &expected, opts) {
+				return
+			}
+			if err := findInOptionList(t, opts, "paths", paths[:i]); err != nil {
+				return
+			}
+		})
+	}
+
+	t.Run("--pid-file=/path/to/foo", func(t *testing.T) {
+		opts, err := c.ParseArgs("ls", "--pid-file=/path/to/foo")
+		if !assert.NoError(t, err, "cli.ParseArgs should succeed") {
+			return
+		}
+
+		expected := options{
+			Command:  "ls",
+			Interval: 1,
+			PidFile:  "/path/to/foo",
+		}
+
+		if !assert.Equal(t, &expected, opts) {
+			return
+		}
+		if err := findInOptionList(t, opts, "pid_file", "/path/to/foo"); err != nil {
+			return
+		}
+	})
+
+	ports := []string{
+		"8080",
+		"0.0.0.0:9090",
+	}
+	for i := 1; i <= 2; i++ {
+		args := make([]string, len(ports))
+		for i, p := range ports {
```

Comment:
No, the outer `i` controls how many arguments `ParseArgs` receives. I suppose there are better ways to write this, especially by removing `i`, but you completely miss the point of the code, and you are just looking at style.